src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php

```diff
@@ -541,25 +542,12 @@ public function testSerializerCacheDisabled()
      */
     public function testDeprecatedSerializerCacheOption()
     {
-        $deprecations = array();
-        set_error_handler(function ($type, $msg) use (&$deprecations) {
-            if (E_USER_DEPRECATED !== $type) {
-                restore_error_handler();
+        ErrorAssert::assertDeprecationsAreTriggered('The "framework.serializer.cache" option is deprecated', function () {
```

Comment:
I'm not sure we should keep this changes now that 3.1 has been released, it may create merge conflicts in the future.